src/resolve.rs

```diff
@@ -14,22 +14,23 @@ use std::{
 };
 
 impl Resolver {
-    pub(crate) fn append_ext_for_path(path: &Path, ext: &str) -> PathBuf {
-        PathBuf::from(&format!("{}{ext}", path.display()))
-    }
-
     fn resolve_file_with_ext(&self, path: PathBuf, info: Info) -> State {
+        let mut path = path.display().to_string();
```

Comment:
I think you should repeatly call `path.set_extension` instead of working on strings.